### PR TITLE
Automate azure resource template deployment

### PR DESCRIPTION
**Pull request by johnfolly**
# Automate azure resource template deployment 

## Usage: `babylon azure arm [OPTIONS] COMMAND [ARGS]`

  Group interacting with Azure Resources Manager

  Requires key `azure_subscription` in the platform config file.

## Options:
  `-h`, `--help`  Show help message and exit.

## Commands:

  `create`  Apply a resource deployment config via arm deployment.
  `delete`  Delete a resource deployment via arm deployment.
  `list`    List all the deployments for a resource group.
  `run`     Apply a resource deployment config via arm deployment.

### CODE REVIEW

File: Babylon/groups/azure/groups/arm/commands/delete.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@pass_arm_client
+@argument("deployment_name")
+@require_platform_key("resource_group_name", "resource_group_name")
+def delete(
+    arm_client: ResourceManagementClient,
+    deployment_name: str,
+    resource_group_name: str,
+) -> Optional[str]:
+    """Delete a resource deployment via arm deployment."""
+
```
Comment (o-reo) [inline Babylon/groups/azure/groups/arm/commands/delete.py:27]: You should use `utils.interactive.confirm_deletion`

File: Babylon/groups/azure/groups/arm/commands/delete.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            deployment_name=deployment_name,
+        )
+    except HttpResponseError as _e:
+        logger.error(f"An error occurred : {_e.message}")
+        return
+
+    logger.debug(poller.result())
+    logger.info(f"Deployment {deployment_name} deleted with status : {poller.status()}")
+
+    return poller.status()
```
Comment (o-reo) [inline Babylon/groups/azure/groups/arm/commands/delete.py:41]: You may rebase to main and use the new `CommandResponse` class !
See  `azure acr get-all` for an example

File: Babylon/groups/azure/groups/arm/commands/run.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        )
+    except HttpResponseError as _e:
+        logger.error(f"An error occurred : {_e.message}")
+        return
+
+    logger.debug(poller.result())
+    logger.info(f"Deployment finished with status : {poller.status()}. \
+                 \nMore details at : {poller.result()['id']}")
+
+    return poller.status()
```
Comment (o-reo) [inline Babylon/groups/azure/groups/arm/commands/run.py:64]: `CommandResponse`

**Review by o-reo — COMMENTED** [review]
:+1:

File: Babylon/utils/configuration.py [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import os
```
Comment (o-reo) [inline Babylon/utils/configuration.py:1]: You are committing something that has already been pushed.
Rebase was not done correctly.

File: Babylon/utils/response.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+from typing import Any
```
Comment (o-reo) [inline Babylon/utils/response.py:1]: Rebase was not done correctly

File: Babylon/utils/working_dir.py [hunk 6]
```diff
@@ -2,7 +2,7 @@
 import pathlib
```
Comment (o-reo) [inline Babylon/utils/working_dir.py:2]: Rebase was not done correctly

**Review by o-reo — CHANGES_REQUESTED** [review]
There are some missing CommandResponse. 
This branch is commiting changes that are already merge, there is something wrong with the way it has been rebased.

**Comment by johnfolly** [comment]
@o-reo you can do a final review now

File: Babylon/groups/azure/groups/arm/commands/create.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        arm_deployment = _commented_yaml_loader.load(file)
+
+    arm_deployment["deployment_name"] = deployment_name
+    arm_deployment["template_uri"] = template_uri
+
+    with open(deployment_name + ".yaml", "w") as _f:
+        _commented_yaml_loader.dump(arm_deployment, _f)
+
+    logger.info(f"Resource deployment config created, content was dumped in {deployment_name}.yaml")
+    return CommandResponse(data=f"Resource deployment config created, content was dumped in {deployment_name}.yaml")
```
Comment (o-reo) [inline Babylon/groups/azure/groups/arm/commands/create.py:44]:
```suggestion
    return CommandResponse(data=arm_deployment)
```
Data must be a `dict`

**Review by o-reo — APPROVED** [review]

**Pull request merged**